### DISCUSSION TITLE
Rich context— without being treated as prompts?

### DISCUSSION THREAD

**Discussion by HenryGetz** [DeabLabs/cannoli#45] [General]
Sorry if I missed this in the documentation, but is there a way to use everything before a content node as context, instead of a call node?

I have complex migration documentation in a canvas, that it would be super helpful to be able to have a call node at the end of it that automatically passed it all as context. Right now it just tries to run everything as a prompt. Thanks!

Here's the canvas so you can see what I mean

![image](https://github.com/user-attachments/assets/f2bebcd1-00dd-4d16-b594-1652049daf67)

**Comment by blindmansion** [comment]
Hey Henry,

Currently we don't have a functionality like this. When you run a canvas as a cannoli, the whole thing will be treated as cannoli logic. However, we are working on a feature which would allow cannolis to use embedded canvas files as context just like markdown files. We'll let you know when that is live!

Thanks for using Cannoli!
